libraries/chain/transaction.cpp

```diff
@@ -504,10 +515,11 @@ packed_transaction::packed_transaction(packed_transaction_v0&& other, bool legac
 packed_transaction_v0_ptr packed_transaction::to_packed_transaction_v0() const {
    const auto* sigs = get_signatures();
    const auto* context_free_data = get_context_free_data();
-   signed_transaction strx( transaction( get_transaction() ),
+
+   return std::make_shared<const packed_transaction_v0>( bytes(packed_trx),
                             sigs != nullptr ? *sigs : vector<signature_type>(),
-                            context_free_data != nullptr ? *context_free_data : vector<bytes>() );
-   return std::make_shared<const packed_transaction_v0>( std::move( strx ), get_compression() );
+                            context_free_data != nullptr ? *context_free_data : vector<bytes>(),
+                            compression.value );
```

Comment:
- You can copy over the `packed_context_free_data` directly. No need to pass in unpacked and then pack. So you can use the existing constructor.
- `bytes(packed_trx)` is not needed, can be just `packed_trx`

We should consider removing the partial support for `full` as we don't actually allow a `full` to be used. Note that `get_prunable_size_impl` for `full` asserts. I think we should modify the constructor of `packed_transaction` that takes a `packed_transaction_v0` to assert of `legacy == false`.

Reply:
Thanks for your suggestion. This simplifies the code. I will do removing the partial support for full later. It did not work in my first try. I'd like to finish other comments first.